genesyscloud/resource_genesyscloud_architect_user_prompt.go

```diff
@@ -0,0 +1,1081 @@
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Computed:    true,
+				ConfigMode:  schema.SchemaConfigModeAttr,
+				Elem:        userPromptResource,
+			},
+		},
+	}
+}
+
+func createUserPrompt(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	architectApi := platformclientv2.NewArchitectApiWithConfig(sdkConfig)
+
+	log.Printf("Creating user prompt %s", name)
+	userPrompt, _, err := architectApi.PostArchitectPrompts(platformclientv2.Prompt{
+		Name:        &name,
+		Description: &description,
+	})
+	if err != nil {
+		return diag.Errorf("Failed to create user prompt %s: %s", name, err)
+	}
+
+	// Create the prompt resources
+	if resources, ok := d.GetOk("resources"); ok && resources != nil {
+		promptResources := resources.(*schema.Set).List()
+		for _, promptResource := range promptResources {
+			resourceMap := promptResource.(map[string]interface{})
+			resourceLanguage := resourceMap["language"].(string)
+
+			tag := make(map[string][]string)
+			tag["filename"] = []string{resourceMap["filename"].(string)}
+
+			promptResource := platformclientv2.Promptassetcreate{
+				Language: &resourceLanguage,
+				Tags:     &tag,
+			}
+
+			resourceTtsString := resourceMap["tts_string"]
+			if resourceTtsString != nil || resourceTtsString.(string) != "" {
+				strResourceTtsString := resourceTtsString.(string)
+				promptResource.TtsString = &strResourceTtsString
+			}
+
+			resourceText := resourceMap["text"]
+			if resourceText != nil || resourceText.(string) != "" {
+				strResourceText := resourceText.(string)
+				promptResource.Text = &strResourceText
+			}
+
+			log.Printf("Creating user prompt resource for language: %s", resourceLanguage)
+			userPromptResource, _, err := architectApi.PostArchitectPromptResources(*userPrompt.Id, promptResource)
+			if err != nil {
+				return diag.Errorf("Failed to create user prompt resource %s: %s", name, err)
+			}
+			uploadUri := userPromptResource.UploadUri
+
+			resourceFilename := resourceMap["filename"]
+			if resourceFilename.(string) == "" {
+				continue
+			}
+			resourceFilenameStr := resourceFilename.(string)
+
+			status, content := uploadPrompt(uploadUri, &resourceFilenameStr, sdkConfig)
+			log.Printf("Status of upload: %s", status)
+			log.Printf("Content of upload: %s", content)
+		}
+	}
+
+	d.SetId(*userPrompt.Id)
+
+	log.Printf("Created user prompt %s %s", name, *userPrompt.Id)
+
+	return readUserPrompt(ctx, d, meta)
+}
+
+func readUserPrompt(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	architectAPI := platformclientv2.NewArchitectApiWithConfig(sdkConfig)
+
+	log.Printf("Reading User Prompt %s", d.Id())
+	userPrompt, resp, getErr := architectAPI.GetArchitectPrompt(d.Id())
```

Comment:
Could you please add some retry to this read, you can look at how I added it to other resources. It avoids issues where the resource is created and the read returns 404 because of how dynamodb eventual consistency works.